python/TestHarness/util.py

```diff
@@ -450,6 +470,46 @@ def deleteFilesAndFolders(test_dir, paths, delete_folders=True):
                     # TL;DR; Just pass...
                     pass
 
+# Check if test has any redirected output, and if its ready to be read
+def checkOutputReady(tester, options):
+    for redirected_file in tester.getRedirectedOutputFiles(options):
+        file_path = os.path.join(tester.getTestDir(), redirected_file)
+        if not os.access(file_path, os.R_OK):
+            return False
+
+    return True
+
+# return concatenated output from tests with redirected output
+def getOutputFromFiles(tester, options):
+    file_output = ''
+    if checkOutputReady(tester, options):
+        for iteration, redirected_file in enumerate(tester.getRedirectedOutputFiles(options)):
+            file_path = os.path.join(tester.getTestDir(), redirected_file)
+            with open(file_path, 'r') as f:
+                file_output += "#"*80 + "\nOutput from processor " + str(iteration) + "\n" + "#"*80 + "\n" + readOutput(f, options)
+    return file_output
+
+# This function reads output from the file (i.e. the test output)
+# but trims it down to the specified size.  It'll save the first two thirds
+# of the requested size and the last third trimming from the middle
+def readOutput(f, options, max_size=100000):
+    first_part = int(max_size*(2.0/3.0))
+    second_part = int(max_size*(1.0/3.0))
+    output = ''
+
+    f.seek(0)
+    if options.sep_files != True:
+        output = f.read(first_part)     # Limit the output to 1MB
+        if len(output) == first_part:   # This means we didn't read the whole file yet
+            output += "\n" + "#"*80 + "\n\nOutput trimmed\n\n" + "#"*80 + "\n"
+            f.seek(-second_part, 2)       # Skip the middle part of the file
+
+            if (f.tell() <= first_part):  # Don't re-read some of what you've already read
+                f.seek(first_part+1, 0)
+
+    output += f.read()              # Now read the rest
+    return output
+
 # See http://code.activestate.com/recipes/576570-dependency-resolver/
 class DependencyResolver:
```

Comment:
Do we still want to have DependencyResolver and ReverseReachability? It doesn't seem we use it anywhere...